app/server/lib/OIDCConfig.ts

```diff
@@ -130,6 +131,10 @@ export class OIDCConfig {
   }
 
   public async getLogoutRedirectUrl(req: express.Request, redirectUrl: URL): Promise<string> {
+    // For IdPs that don't have end_session_endpoint, we just redirect to the logout page.
+    if (this._client.issuer.metadata.end_session_endpoint === undefined) {
```

Comment:
I am a little wary being lax about logout, since sometimes strict predictable behavior could be important. What do you think of controlling it using an env var: if `GRIST_OIDC_IDP_SKIP_END_SESSION` is set, then skip the `endSessionUrl()` call, otherwise, check that `end_session_endpoint` is present at initialization and throw an exception if it's not (with a pointer to the env var for how to skip it)?

Reply:
An example of the danger of skipping IdP is for a configuration where the IdP is just there to serve Grist. If we skip IdP on logout, then a user would click "logout" and the app would look logged out, but clicking "log in" again would redirect through IdP, which is still logged in, and sign the user in without asking for a password. In other words, the user may think they are signed out even though anyone using this browser can still access all their Grist data by just clicking "log in" again.

So I think it's important that we don't fall back to this configuration as a default, but that the user setting it up selects it explicitly.

(It's possible I misunderstand the implications, but that kind of scenario is my concern.)

Reply:
OK, I added the env variable as requested :)